projects/worker-cc-photoshop/actions/worker-cc-photoshop/index.js

```diff
@@ -51,11 +53,30 @@ function getAuthorization(params) {
  * @param {Object} instructions Rendition instructions
  * @returns {Object} options for Photoshop Actions Api request
  */
-async function setupPhotoshopActionsOptions(client, instructions) {
+async function setupPhotoshopActionsOptions(client, instructions, files) {
     if (!instructions || !instructions.photoshopAction) {
         throw Error("Photoshop Action url not provided");
     }
-    const options = await client.fileResolver.resolveInputsPhotoshopActionsOptions({ actions: instructions.photoshopAction });
+    
+    // workaround for action files from AEM
+    // we must download the action file and add the `.atn`
+    // extension to the file so Photoshop Service can
+    // recognize it
+    let photoshopAction = instructions.photoshopAction;
+    let ext;
+        try {
+            ext = path.extname(photoshopAction).substring(1).toLowerCase();
+        } catch (err) {
+        }
+        if (!ext) {
+            const tempActionFilename = `${uuidv4()}_temp.atn`;
+            const aioLibActionFilename = `${uuidv4()}/photoshopaction.atn`;;
+            await downloadFile(photoshopAction, tempActionFilename);
+            await files.copy(tempActionFilename, aioLibActionFilename, { localSrc: true });
+            photoshopAction = aioLibActionFilename;
+            
+        }
+    const options = await client.fileResolver.resolveInputsPhotoshopActionsOptions({ actions: photoshopAction });
```

Comment:
Clients should not call `fileResolver` directly, this is an implementation detail of the library. See: https://github.com/adobe/aio-lib-photoshop-api/blob/master/src/index.js#L519

Just pass in the `aioLibActionFilename` in the `href` property.

Reply:
Created https://github.com/adobe/aio-lib-photoshop-api/issues/8 and https://github.com/adobe/aio-lib-photoshop-api/issues/8

Reply:
are you sure this is not ok to use?
`FileResolver` (along with `resolveInputsPhotoshopActionsOptions`) are defined in the readme: https://github.com/adobe/aio-lib-photoshop-api#fileresolverresolveinputsphotoshopactionsoptionsoptions--applyphotoshopactionsoptions

Reply:
if I pass in `aioLibActionFilename` in the `href` property, it will default to `external` storage type right?

Reply:
I see now you linked the wrong issue. https://github.com/adobe/aio-lib-photoshop-api/issues/9 this one to request to make fileresolver private

Reply:
Thanks @jdelbick !  I added FileResolver to the jsdoc to give more information to the library users how the input and outputs are resolved, however it is misleading since it's all done internally and shouldn't be done by clients of the library.